test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -81,11 +81,22 @@ public static TheoryData MultiAttributeCheckBoxData
             string expectedAttributeString)
         {
             // Arrange
-            var originalContent = "original content";
```

Comment:
The changes to this file are very hard to follow. I _think_ you're adding additional checks of `PreElement` and so on but don't know why.

Please revert as much as possible that's not directly related to the `InputTagHelper` changes i.e. don't do much more than add `expectedAttributes` (perhaps pass in the `expectedAttributes`) and split `expectedPostElement` from `expectedContent`.

Alternatively, let's sit down tomorrow and discuss why the new code is an improvement.